### PR TITLE
MAT-6660 add test case configuration

### PR DESCRIPTION
**Pull request by chubert-sb**
## MADiE PR

Jira Ticket: [MAT-6660](https://jira.cms.gov/browse/MAT-6660)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

**Comment by chubert-sb** [comment]
Screenshot 2024-02-28 at 3 10 09 PM

File: src/components/routes/qdm/SDEPage.tsx [hunk 1]
```diff
@@ -1,6 +1,205 @@
...
+            ]}
+            value={formik.values.sdeIncluded}
+            onChange={(e) => {
+              const nextSdeIncluded = e.target.value;
+              formik.setFieldValue("sdeIncluded", nextSdeIncluded);
+            }}
+            error={
+              formik.touched.sdeIncluded && Boolean(formik.errors.sdeIncluded)
+            }
+            helperText={
```
Comment (RohitKandimalla) [inline src/components/routes/qdm/SDEPage.tsx]: Do we need "error" and "helperText" properties ? I mean is there a way to trigger errors in Radio Group ?

File: src/components/routes/qdm/SDEPage.tsx [hunk 2]
```diff
@@ -1,6 +1,205 @@
...
+    }
+  }, [measure]);
+
+  const formik = useFormik({
+    initialValues: {
+      sdeIncluded:
+        String(measure?.testCaseConfiguration?.sdeIncluded) || "true",
+    },
+    enableReinitialize: true,
+    onSubmit: async (values: SDEForm) => await handleSubmit(),
```
Comment (RohitKandimalla) [inline src/components/routes/qdm/SDEPage.tsx]: Values are ignored, Hoping its intentional

File: src/components/routes/qdm/SDEPage.tsx [hunk 3]
```diff
@@ -1,6 +1,205 @@
...
+  useEffect(() => {
+    const subscription = measureStore.subscribe(setMeasure);
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+
+  useEffect(() => {
+    if (measure && measure.testConfiguration) {
+      setCurrentSdeIncluded(measure.testCaseConfiguration.sdeIncluded);
```
Comment (RohitKandimalla) [inline src/components/routes/qdm/SDEPage.tsx]: Why do we need to go through an extra step of having our own state for currentSdeIncluded when formik already has it ?

File: src/components/routes/qdm/SDEPage.tsx [hunk 4]
```diff
@@ -1,6 +1,205 @@
...
+      onSubmit={handleSubmit}
+      data-testid={`sde-form`}
+      style={{ minHeight: 539 }}
+    >
+      <div className="content">
+        <div className="subTitle">
+          <h2>SDE</h2>
+        </div>
+        <div className="sde-page" data-testId="sde-page">
+          <RadioButton
```
Comment (RohitKandimalla) [inline src/components/routes/qdm/SDEPage.tsx:118]: We may want to make sure who can make changes to this form.

File: src/components/routes/qdm/SDEPage.tsx [hunk 5]
```diff
@@ -1,6 +1,205 @@
...
+  useEffect(() => {
+    if (measure && measure.testConfiguration) {
+      setCurrentSdeIncluded(measure.testCaseConfiguration.sdeIncluded);
+    }
+  }, [measure]);
+
+  const formik = useFormik({
+    initialValues: {
+      sdeIncluded:
+        String(measure?.testCaseConfiguration?.sdeIncluded) || "true",
```
Comment (RohitKandimalla) [inline src/components/routes/qdm/SDEPage.tsx]: not sure if this condition works here, if measure.testCaseConfiguration.sdeIncluded is undefined, wouldn't formik.values.sdeIncluded be undefined ?

Also if undefined or null, shouldn't the default value be false ?

File: src/components/routes/qdm/SDEPage.tsx (hunk 1 above)
Comment (chubert-sb) [reply src/components/routes/qdm/SDEPage.tsx]: good point. i took it out

File: src/components/routes/qdm/SDEPage.tsx (hunk 2 above)
Comment (chubert-sb) [reply src/components/routes/qdm/SDEPage.tsx]: Removed

File: src/components/routes/qdm/SDEPage.tsx (hunk 3 above)
Comment (chubert-sb) [reply src/components/routes/qdm/SDEPage.tsx]: removed

File: src/components/routes/qdm/SDEPage.tsx (hunk 4 above)
Comment (chubert-sb) [reply src/components/routes/qdm/SDEPage.tsx:118]: I added the canEdit tag

File: src/components/routes/qdm/SDEPage.tsx (hunk 5 above)
Comment (chubert-sb) [reply src/components/routes/qdm/SDEPage.tsx]: I changed the default to false

**Review by RohitKandimalla — DISMISSED** [review]

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**